packages/server/src/api/controllers/pools.ts

```diff
@@ -0,0 +1,116 @@
+import { celebrate, Joi, Segments } from 'celebrate';
+import {
+  endOfDay,
+  endOfHour,
+  startOfDay,
+  startOfHour,
+  subWeeks,
+} from 'date-fns';
+import { Request, Response, Router } from 'express';
+
+import { dayMs } from 'util/date'
+import { HTTPError } from 'api/util/errors';
+import { isValidEthAddress } from 'util/eth';
+import cacheMiddleware from 'api/middlewares/cache';
+import UV3Fetchers from 'services/uniswap-v3/fetchers';
+import wrapRequest from 'api/util/wrap-request';
+
+// TODO: move this somewhere else, maybe to the fetchers manager.
+// Will also need to namespace by network
+import { wrapWithCache, keepCachePopulated } from 'util/redis-data-cache';
+import redis from 'util/redis';
+const fetcher = UV3Fetchers.get('mainnet');
+
+// TODO: Not actually cached for now, want to fix type of wrapWithCache first
+const cached = {
+  getEthPrice: fetcher.getEthPrice,
+  getTopPools: fetcher.getTopPools,
+  getHistoricalDailyData: fetcher.getHistoricalDailyData,
+  getHistoricalHourlyData: fetcher.getHistoricalHourlyData,
+
+}
+
+// GET /ethPrice
+async function getEthPrice() {
+  const ethPrice = await cached.getEthPrice();
+  return ethPrice;
+}
+
+// GET /pools
+// should gen the query types from the joi schema?
+type GetTopPoolsQuery = { count: number };
+const getTopPoolsValidator = celebrate({
+  [Segments.QUERY]: Joi.object().keys({
+    count: Joi.number()
+  })
+});
+
+async function getTopPools(req: Request<unknown, unknown, unknown, GetTopPoolsQuery>) {
+  const { count } = req.query;
+  return cached.getTopPools(count);
+}
+
+// GET /pools/:id
+async function getPoolOverview(req: Request) {
+  const { id } = req.params;
+  validateEthAddress(id);
```

Comment:
Is there a way to make this a custom validator at the `Joi` level?

Reply:
Looks like yes: https://github.com/sideway/joi/blob/master/API.md#anycustommethod-description